### PR TITLE
Feat auth ext

### PR DESCRIPTION
**Pull request by Nankun-Xu**
## Describe your changes
Create google/facebook/apple sign in logic

## Issue ticket number and link
#25 

## Screenshots / Video

https://github.com/user-attachments/assets/2e1039b1-81cd-4fc8-a120-861e3c20339c

## Checklist
- [x] I have performed a self-review of my code
- [x] I have used `npx expo-doctor` to review any dependency issues.
- [x] I have pulled and merged any recent changes from the main branch (no merge conflicts)
- [x] I have tested thoroughly any edge cases that might break the application
- [x] I have deleted any console.log statements and useless comments

### CODE REVIEW

**Review by Lyton505 — APPROVED** [review]

File: components/auth/Signup.tsx
Comment (dave-cao) [inline components/auth/Signup.tsx:1]: Why is this Signup.tsx here? This should be deleted.

File: components/auth/buttons/GoogleSigninButton.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  GoogleSignin,
+  GoogleSigninButton,
+  statusCodes,
+} from "@react-native-google-signin/google-signin";
+import { supabase } from '@/lib/supabase';
+
+// Configuration
+GoogleSignin.configure({
+  webClientId: "64478294171-estk2fbh8o55du2uidrt6l2h4s10vn8v.apps.googleusercontent.com", // client ID of type WEB for your server. Required to get the `idToken` on the user object, and for offline access.
+  scopes: ["https://www.googleapis.com/auth/drive.readonly"], // what API you want to access on behalf of the user, default is email and profile
```
Comment (dave-cao) [inline components/auth/buttons/GoogleSigninButton.tsx:12]: Do we need drive access?

File: components/auth/buttons/GoogleSigninButton.tsx
Comment (dave-cao) [inline components/auth/buttons/GoogleSigninButton.tsx:1]: Do we even use this google sign in button? If not, we should delete it.

**Comment by dave-cao** [comment]
For google signin, I am getting the popup, but I am getting
image

For facebook I am getting
image

Can't test apple without ios. I tried on my Ipad but it crashed because I needed a build.

**Comment by dave-cao** [comment]
Btw, I rebased this branch with the new changes from main!

**Comment by dave-cao** [comment]
I will talk about this later on Tuesday, but I'm thinking perhaps we keep this branch on standby and continue working on the other features because this only works when we create a build for ios/android. So if we merge this branch, we won't be able to give the client our preview builds. We can just continously keep this branch up to date with main, and merge it at the end. What do you guys think?

@Nankun-Xu @Lyton505

**Pull request closed without merging**